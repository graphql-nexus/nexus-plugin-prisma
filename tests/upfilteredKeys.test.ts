import * as Nexus from 'nexus'
import { generateSchemaAndTypes } from './__utils'

const testData = {
  datamodel: `
  model User {
    id  Int @id
    name String
    nested Nested
  }

  model Nested {
    id Int @id
    name String
  }
`,
  query: Nexus.queryType({
    definition(t: any) {
      t.crud.user()
    },
  }),
  mutation: Nexus.mutationType({
    definition(t: any) {
      t.crud.createOneUser({
        upfilteredKey: 'create',
      })
    },
  }),
  user: Nexus.objectType({
    name: 'User',
    definition: (t: any) => {
      t.model.id()
      t.model.name()
      t.model.nested()
    },
  }),
  nested: Nexus.objectType({
    name: 'Nested',
    definition: (t: any) => {
      t.model.id()
      t.model.name()
    },
  }),
}

it('filters upfilteredKeys', async () => {
  const { datamodel, ...resolvers } = testData
  const { schema } = await generateSchemaAndTypes(
    datamodel,
    Object.values(resolvers),
  )
  expect(schema).toMatchSnapshot('upfilteredKeys')
})
