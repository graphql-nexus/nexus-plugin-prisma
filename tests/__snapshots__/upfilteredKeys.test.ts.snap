// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upfilteredKeys generates new types without upfilteredKeys: upfilteredKeys 1`] = `
"type Mutation {
  createOneUser(data: UserCreateInput!): User!
}

type Nested {
  id: Int!
  name: String!
}

input NestedCreateManyWithoutUserInput {
  create: [NestedCreateWithoutUserInput!]
  connect: [NestedWhereUniqueInput!]
}

input NestedCreateWithoutUserInput {
  id: Int!
  name: String!
}

input NestedWhereUniqueInput {
  id: Int
}

type Query {
  user(where: UserWhereUniqueInput!): User
}

type User {
  id: Int!
  name: String!
  nested(skip: Int, after: Int, before: Int, first: Int, last: Int): [Nested!]!
}

input UserCreateInput {
  id: Int!
  name: String!
  nested: NestedCreateManyWithoutUserInput
}

input UserWhereUniqueInput {
  id: Int
}
"
`;
